tests/test_rabfe.py

```diff
@@ -21,6 +27,193 @@
+
+        ff_params = hif2a_ligand_pair.ff.get_ordered_params()
+
+        temperature = 300.0
+        pressure = 1.0
+        dt = 2.5e-3
+
+        client = CUDAPoolClient(NUM_GPUS)
+
+        decouple_model = RelativeBindingModel(
+            client,
+            hif2a_ligand_pair.ff,
+            host_system,
+            construct_lambda_schedule(2),
+            host_topology,
+            temperature,
+            pressure,
+            dt,
+            10,
+            50,
+            frame_filter=all_frames,
+        )
+
+        blocker = hif2a_ligand_pair.mol_a
+        ligand = hif2a_ligand_pair.mol_b
+
+        decouple_topo = decouple_model.setup_topology(blocker, ligand)
+        decouple_ref = RelativeFreeEnergy(decouple_topo)
+
+        decouple_unbound_potentials, decouple_sys_params, _ = decouple_ref.prepare_host_edge(
+            ff_params, decouple_model.host_system
+        )
+
+        conv_model = AbsoluteConversionModel(
+            client,
+            hif2a_ligand_pair.ff,
+            host_system,
+            construct_lambda_schedule(2),
+            host_topology,
+            temperature,
+            pressure,
+            dt,
+            10,
+            50,
+            frame_filter=all_frames,
+        )
+
+        conv_topo = conv_model.setup_topology(ligand)
+        conv_ref = AbsoluteFreeEnergy(ligand, conv_topo)
+
+        conv_unbound_potentials, conv_sys_params, _ = conv_ref.prepare_host_edge(ff_params, conv_model.host_system)
+
+        assert len(conv_sys_params) == len(decouple_sys_params)
+        seen_nonbonded = False
+        for i, decouple_pot in enumerate(decouple_unbound_potentials):
+            if not isinstance(decouple_pot, NonbondedInterpolated):
+                continue
+            seen_nonbonded = True
+
+            conv_pot = conv_unbound_potentials[i]
+            assert isinstance(conv_pot, NonbondedInterpolated)
+
+            conv_nonbonded_params = conv_sys_params[i]
+            decouple_nonbonded_params = decouple_sys_params[i]
+
+            # Should have the same number of parameters besides the blocker. Since params are interpolated, multiply by 2
+            assert conv_nonbonded_params.shape[0] == decouple_nonbonded_params.shape[0] - blocker.GetNumAtoms() * 2
+            # Should both share the same number of parameters types
+            assert conv_nonbonded_params.shape[1] == decouple_nonbonded_params.shape[1]
+
+            conv_params = conv_nonbonded_params[num_host_atoms * 2 :]
+            decouple_params = decouple_nonbonded_params[num_host_atoms * 2 :]
+
+            assert conv_params.shape[0] == decouple_params.shape[0] - blocker.GetNumAtoms() * 2
+
+            # Verify the dest params of conv match the src params of decouple
+            np.testing.assert_array_equal(
+                conv_params[len(conv_params) // 2 :],
+                decouple_params[len(decouple_params) // 2 + blocker.GetNumAtoms() :],
+            )
```

Comment:
Verified some of these index calculations but ran out of time with others
* `conv_params` is assumed concatenated in the order `[src, dest]`, verified consistent with [definition in BaseTopologyConversion](https://github.com/proteneer/timemachine/blob/06e2512425e70278538f1e440e9f5dd83b3af4af/fe/topology.py#L364)
* `decouple_params` is assumed concatenated in the order `[dest, blocker, src]`, but I'm having trouble inferring that from the definitions in [DualTopologyStandardDecoupling](https://github.com/proteneer/timemachine/blob/06e2512425e70278538f1e440e9f5dd83b3af4af/fe/topology.py#L601) or [HostGuestTopology](https://github.com/proteneer/timemachine/blob/06e2512425e70278538f1e440e9f5dd83b3af4af/fe/topology.py#L201-L255)

Strongly prefer if alchemical topology classes provided more hint of the memory layout that users must be aware of -- maybe named index arrays `top.host_idxs`, `top.mol_a_idxs`,  etc. -- will move this comment to https://github.com/proteneer/timemachine/issues/565

Reply:
Will add a comment to explain it a bit more, as I think this is useful context